src/conv/invokers/impl_gemm_dynamic.cpp

```diff
@@ -505,10 +509,87 @@ InvokerFactory MakeImplGemmDynamicForwardXdlopsNHWCInvokerFactory(
     const auto isGfx90aFp16altSupport =
         (ctx.GetStream().GetDeviceName() == "gfx90a") && conv_problem.IsFp16();
 
+    const bool need_cast = [&]() {
+        if(ctx.conv_problem.GetOut().GetType() == miopenHalf)
+            return use_fp32_global_split_on_fp16;
+        if(ctx.conv_problem.GetOut().GetType() == miopenBFloat16)
+            return need_set_zero;
+        return false;
+    }();
+    const auto isNCHW = ctx.IsLayoutDefault();
+
+    size_t trans_input_offset = 0;
+    size_t trans_input_size   = 0;
+
+    size_t trans_weight_offset = 0;
+    size_t trans_weight_size   = 0;
+
+    size_t trans_output_offset = 0;
+    size_t trans_output_size   = 0;
+
+    bool trans_input_skippable  = false;
+    bool trans_weight_skippable = false;
+    bool trans_output_skippable = false;
+
+    int trans_input_idx  = -1;
+    int trans_weight_idx = -1;
+    int trans_output_idx = -1;
+
+    if(isNCHW)
+    {
+        TransposeSolution_NCHW2NHWC trans_input(ctx, ctx.in_data_type, n, c, hi, wi);
+        TransposeSolution_NCHW2NHWC trans_weight(ctx,
+                                                 ctx.weights_data_type,
+                                                 k,
+                                                 c / group,
+                                                 y,
+                                                 x); // group * k_per_group as batch for weight
+        TransposeSolution_NHWC2NCHW trans_output(ctx, ctx.out_data_type, n, k, ho, wo);
+
+        trans_input_skippable  = trans_input.IsSkippable();
+        trans_weight_skippable = trans_weight.IsSkippable();
+        trans_output_skippable = trans_output.IsSkippable();
+
+        if(!trans_input_skippable)
+            opArgsTrans.emplace_back(trans_input.GetKernelArg());
+        if(!trans_weight_skippable)
+            opArgsTrans.emplace_back(trans_weight.GetKernelArg());
+        if(!trans_output_skippable)
+            opArgsTrans.emplace_back(trans_output.GetKernelArg());
+
+        trans_input_size  = trans_input_skippable ? 0 : trans_input.GetSize();
+        trans_weight_size = trans_weight_skippable ? 0 : trans_weight.GetSize();
+        trans_output_size = trans_output_skippable ? 0 : trans_output.GetSize();
+
+        trans_weight_offset = trans_input_offset + trans_input_size;
+        trans_output_offset = trans_weight_offset + trans_weight_size;
+
+        int idx = 0;
+        if(!trans_input_skippable)
+            trans_input_idx = idx++;
+        if(!trans_weight_skippable)
+            trans_weight_idx = idx++;
+        if(!trans_output_skippable)
+            trans_output_idx = idx++;
+    }
+
+    const size_t cast_offset = isNCHW ? (trans_output_offset + trans_output_size) : 0;
+    const size_t cast_size =
+        need_cast ? static_cast<size_t>(n) * k * ho * wo * miopen::GetTypeSize(miopenFloat) : 0;
+
+    const int kID_trans_start = isGfx90aFp16altSupport ? 2 : 1;
```

Comment:
This is wrong. Kernel index shall be computed within Invoker. Otherwise it will remain the same regardless of the ALT attribute.

Reply:
OK. Let me use the same method in the invoker to pick up the kernel index

Reply:
Let me recheck. It seems like **_I_** was wrong.

Reply:
I withdraw my original comment. The code is correct.